Fix scene edit state resetting when scene is updated

InfiniteTF opened the pull request:
Replaces useEffect form initialization with initialization directly in the state variables, so state is only ever initialized once. This avoids the issue where updating organized/ocounter state would reinitialize the form.

Resolves #1110.

WithoutPants commented:
This works, but could you also cover the Image, Gallery, and Performer pages which have the same bug?

InfiniteTF commented:
I've fixed the gallery and image pages. The performer page really needs a refactor, but I fixed the favorite button resetting state by removing the loading state, which I think is a better experience anyway.
